src/go/web/handlers.go

```diff
@@ -3332,6 +3338,214 @@ func GetError(w http.ResponseWriter, r *http.Request) {
+		return
+	}
+
+	activeMounts[mm.GetLocalMountPath(vars["exp"], vars["name"])] = struct{}{}
+	
+	w.WriteHeader(http.StatusOK)
+}
+
+// POST /experiments/{exp}/vms/{name}/unmount
+func UnmountVM(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+
+	role := r.Context().Value("role").(rbac.Role)
+	if !role.Allowed("vms/mount", "patch") {
+		http.Error(w, "forbidden", http.StatusForbidden)
+		return
+	}
+
+	mountsMu.Lock()
+	defer mountsMu.Unlock()
+	_, err := mm.ExecC2Command(mm.C2NS(vars["exp"]), mm.C2VM(vars["name"]), mm.C2Unmount(), mm.C2IDClientsByUUID())
+
+	if err != nil {
+		log.Error(err.Error())
+		http.Error(w, fmt.Sprintf("Error unmounting: %s", err), http.StatusInternalServerError)
+		return
+	}
+
+	delete(activeMounts, mm.GetLocalMountPath(vars["exp"], vars["name"]))
+
+	w.WriteHeader(http.StatusOK)
+}
+
+// GET /experiments/{exp}/vms/{name}/mount/files?path=
+func GetMountFiles(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	basePath := mm.GetLocalMountPath(vars["exp"], vars["name"])
+
+	role := r.Context().Value("role").(rbac.Role)
+	if !role.Allowed("vms/mount", "patch") {
+		http.Error(w, "forbidden", http.StatusForbidden)
+		return
+	}
+
+	mountsMu.RLock()
+	defer mountsMu.RUnlock()
+	_, exists := activeMounts[basePath]
+
+	if !exists {
+		http.Error(w, fmt.Sprintf("No existing mount for %s", basePath), http.StatusBadRequest)
+		return
+	}
+
+	combinedPath := filepath.Join(basePath, vars["path"])
+	log.Info("combinedPath: %s", combinedPath)
+	fileInfo, err := os.Stat(combinedPath)
+
+	if err != nil {
+		errString := fmt.Sprintf("Error getting path %s: %s", combinedPath, err.Error())
+		log.Error(errString)
+		http.Error(w, errString, http.StatusInternalServerError)
+		return
+	}
+	if !strings.HasPrefix(combinedPath, basePath) {
+		errString := fmt.Sprintf("Error getting path %s: Path is not within mount", combinedPath)
+		log.Error(errString)
+		http.Error(w, errString, http.StatusBadRequest)
+		return
+	}
+	if !fileInfo.IsDir() {
+		http.Error(w, fmt.Sprintf("Expected directory not file: %s", combinedPath), http.StatusBadRequest)
+		return
+	}
+
+	dirFiles, err := ioutil.ReadDir(combinedPath)
+	if err != nil {
+		errString := fmt.Sprintf("Error getting files in %s: %s", combinedPath, err.Error())
+		log.Error(errString)
+		http.Error(w, errString, http.StatusInternalServerError)
+	}
+
+	var files file.Files
+	for _, f := range dirFiles {
+		file := file.MakeFile(f, combinedPath)
+		file.Path = strings.TrimPrefix(file.Path, basePath)
+		files = append(files, file)
+	}
+	body, _ := json.Marshal(util.WithRoot("files", files))
+	w.Write(body)
+}
+
+// GET /experiments/{exp}/vms/{name}/files/download?path=
+func DownloadMountFile(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	basePath := mm.GetLocalMountPath(vars["exp"], vars["name"])
+
+	role := r.Context().Value("role").(rbac.Role)
+	if !role.Allowed("vms/mount", "patch") {
```

Comment:
Should use `get` instead of `patch` here since we're downloading a specific file from a mount, and should pass `{exp_name}/{vm_name}` to `role.Allowed` as well to support per-VM access control for mounts.

Reply:
Thanks for this. I wasn't really familiar with the RBAC system. 

One open question I need to figure out is that right now `GetMountFiles` only requires `list` permissions, but for that method to work, you'll have to have mounted the vm which requires `post`. 

My original thought was that only users should be able to use the feature, but maybe viewers should be able to create a mount and view files as they are. Any thoughts @aherna ?

Reply:
We can update the RBAC policies for the default roles we include in phenix to adjust for this. I agree that viewers should be able to mount and view files, and perhaps download, but not upload them.

As for the need to have both `post` and `list` permissions to actually list files, we can handle that in the policy config.

Reply:
Alright, I think I've got this mostly done, though since I still don't fully know the RBAC system, I may have missed some stuff for the experiment/vm viewers.

Reply:
Yeah, we can tease the RBAC stuff out as we go. The RBAC stuff on the UI side needs to be overhauled, since right now it's limited to the predefined roles. I know how I want to refactor it, just haven't had the time to do so yet.